libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -338,12 +337,14 @@ func (dhtClient *DHTClient) newStreamLoopUntilTimeout(
 			stream, err = dhtClient.routedHost.NewStream(ctx, peerID, streamType)
 		case <-ctx.Done():
 			sleepTime = 0
-			break
 		}
 		if sleepTime == 0 {
 			break
 		}
 	}
+	if stream == nil {
+		return stream, errors.New("stream nil" + err.Error())
+	}
```

Comment:
New check!

Reply:
return nil, errors.New("stream nil" + err.Error())

Reply:
or set `err=errors.New("Can not get a stream within timeout")`, instead of break

Reply:
stream is nil per check